### ISSUE TITLE
Does (horovod + learning rate decay) work properly?

### ISSUE THREAD

**Issue by wcshin-git** [lucidrains/DALLE-pytorch#268]
Hi,
I'm using Horovod on 3 GPUs(in a single machine) with learning rate decay`ReduceLROnPlateau`.
But I found something strange while looking at the printed log.
![image](https://user-images.githubusercontent.com/76698585/120114327-a7027a00-c1b9-11eb-8ebe-8120a5f0ebea.png)
![image](https://user-images.githubusercontent.com/76698585/120114349-c26d8500-c1b9-11eb-8760-99ac269005cc.png)
It seems that the learning rate for each process(GPU) is not synchronized.

Actually, this is my first time using Horovod and I don't know what's going on inside Horovod. I would appreciate it if you could let me know if I am mistaken!

**Comment by afiaka87** [comment]
Good catch! That does appear to be a bug. Unfortunately, not a lot of us are using horovod these days now that DeepSpeed seems to  have stabilized a good deal and trains much faster. 

Have you tried using DeepSpeed with the Dockerfile in the repository?

**Comment by janEbert** [comment]
Hey, thanks for the report, that's a great catch! Passing `avg_loss` instead of `loss` into the scheduler should already fix this! Sorry for the issue. :)
That way, all processes use the same loss, so the scheduler stays the same across all workers.

**Comment by janEbert** [comment]
Would you like to PR the fix?

**Comment by wcshin-git** [comment]
@afiaka87 @janEbert  Thanks for your reply!
Ok I'll PR the fix!  
@janEbert you mean, 
BEFORE: 
```
# at 447-th line in train_dalle.py
    if LR_DECAY and not using_deepspeed:
        # Scheduler is automatically progressed after the step when
        # using DeepSpeed.
        distr_scheduler.step(loss)
```
AFTER:
```
    if LR_DECAY and not using_deepspeed:
        # Scheduler is automatically progressed after the step when
        # using DeepSpeed.
        distr_scheduler.step(avg_loss)
```
If this is right, I'll PR the fix right now:)

**Comment by janEbert** [comment]
Yeah, exactly! Sorry I wasn't clearer in my description. Thank you!
